lib/praxis-mapper/connection_factories/sequel.rb

```diff
@@ -0,0 +1,57 @@
+module Praxis::Mapper
+  module ConnectionFactories
+
+    class Sequel
+
+      def initialize(connection:nil, **opts)
+        raise ArgumentError, 'May not provide both a connection and opts' if connection && !opts.empty?
+
+        if connection
+          @connection = connection
+        else
+          @connection = ::Sequel.connect(**opts)
+        end
+
+        @timeout = @connection.pool.instance_variable_get(:@timeout)
+        @sleep_time = @connection.pool.instance_variable_get(:@sleep_time)
+
+        # connections that we created explicitly
+        @owned_connections = Hash.new
+      end
+
+      def checkout(connection_manager)
+        unless acquire(connection_manager.thread)
+          time = Time.now
+          timeout = time + @timeout
+          sleep_time = @sleep_time
+          sleep sleep_time
+          until acquire(connection_manager.thread)
+            raise(::Sequel::PoolTimeout) if Time.now > timeout
+            sleep sleep_time
+          end
+        end
+
+        @connection
+      end
+
+      def release(connection_manager, connection)
```

Comment:
Let's add some comments:
Make sure we only release a connection that this repository acquired. 
This is specially tricky since this same thread could already have acquired the connection from Sequel directly, and we don't want to release that underlying connection if that's the case. That's why we need to keep track of our "owned_connections".
This way, we allow external management of Sequel connections unaffected (even for the same thread and transaction blocks)